src/components/ChannelPreview/ChannelPreview.tsx

```diff
@@ -109,10 +110,25 @@ export const ChannelPreview = <
     };
   }, [refreshUnreadCount, channelUpdateCount]);
 
+  useEffect(() => {
+    const handleEvent = () => {
+      const newDisplayTitle = getDisplayTitle(channel, client.user);
+      const newDisplayImage = getDisplayImage(channel, client.user);
+      if (newDisplayTitle !== displayTitle) {
+        setDisplayTitle(newDisplayTitle);
+      }
+      if (newDisplayImage !== displayImage) {
+        setDisplayImage(newDisplayImage);
+      }
```

Comment:
Let's not rely on "outside" values and rather use previous value inside setState action:
```tsx
setDisplayTitle((previousValue) => previousValue !== newValue ? newValue : previousValue);
```
Returning the same value won't trigger the rerender as [stated here](https://reactjs.org/docs/hooks-reference.html#bailing-out-of-a-state-update).

Reply:
@arnautov-anton thank you for pointing this out. I have updated the snippet.